### PR TITLE
[Network] Fix cbor decoder validation

### PR DESCRIPTION
**Pull request by peterargue**
Fixes data validation for cbor encoding and adds unit test coverage

### CODE REVIEW

**Review by vishalchangrani — APPROVED** [review]

File: network/codec/cbor/decoder_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		assert.Nil(t, decoded)
+		assert.ErrorIs(t, err, codec.ErrInvalidEncoding)
+	})
+
+	t.Run("returns error when message code is invalid", func(t *testing.T) {
+		var buf bytes.Buffer
+
+		// code == min
+		err := cborcodec.NewCodec().NewEncoder(&buf).Encode([]byte{codec.CodeMin})
+		require.NoError(t, err)
```
Comment (yhassanzadeh13) [inline network/codec/cbor/decoder_test.go:83]: Please explain that we expect the first byte to be the message type code.

**Review by yhassanzadeh13 — APPROVED** [review]

File: network/codec/cbor/codec_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/network/codec"
+	"github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestCodecDecode(t *testing.T) {
```
Comment (kc1116) [inline network/codec/cbor/codec_test.go:15]: add t.Parallel()

Comment (kc1116) [inline network/codec/cbor/codec_test.go:15]:
```suggestion
func TestCodec_Decode(t *testing.T) {
```

File: network/codec/cbor/decoder_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	"github.com/stretchr/testify/require"
+
+	cborcodec "github.com/onflow/flow-go/model/encoding/cbor"
+	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/network/codec"
+	"github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestDecoderDecode(t *testing.T) {
```
Comment (kc1116) [inline network/codec/cbor/decoder_test.go:17]:
```suggestion
func TestDecoder_Decode(t *testing.T) {
```

File: network/codec/errors.go [hunk 4]
```diff
@@ -5,6 +5,9 @@ import (
 	"fmt"
 )
 
+// ErrInvalidEncoding is returned when attempting to decoded a message with an invalid encoding.
+var ErrInvalidEncoding = errors.New("invalid encoding")
```
Comment (kc1116) [inline network/codec/errors.go:9]: Can we make this a little more descriptive or is it generic on purpose ?

**Review by kc1116 — APPROVED** [review]
Left a couple small comments otherwise LGTM 🤙

File: network/codec/errors.go (hunk 4 above)
Comment (peterargue) [reply network/codec/errors.go]: addressed in 9c0fb6f

File: network/codec/cbor/decoder_test.go (hunk 3 above)
Comment (peterargue) [reply network/codec/cbor/decoder_test.go]: addressed in 9c0fb6f

File: network/codec/cbor/codec_test.go (hunk 2 above)
Comment (peterargue) [reply network/codec/cbor/codec_test.go]: addressed in 9c0fb6f

Comment (peterargue) [reply network/codec/cbor/codec_test.go]: addressed in 9c0fb6f

File: network/codec/cbor/decoder_test.go (hunk 1 above)
Comment (peterargue) [reply network/codec/cbor/decoder_test.go]: addressed in 9c0fb6f

File: network/codec/cbor/decoder_test.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+		require.Equal(t, blockProposal, decoded)
+	})
+
+	t.Run("returns error when data is empty", func(t *testing.T) {
+		var buf bytes.Buffer
+		// empty buffer
+
+		decoded, err := c.NewDecoder(&buf).Decode()
+		assert.Nil(t, decoded)
+		assert.ErrorContains(t, err, "could not decode message")
```
[the remark this answers is not in the document]
Comment (peterargue) [reply network/codec/cbor/decoder_test.go]: addressed in 9c0fb6f

**Comment by peterargue** [comment]
bors merge

**Pull request merged**